packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/ec2-run-task.ts

```diff
@@ -0,0 +1,105 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import { Construct } from '@aws-cdk/core';
+import { EcsRunTaskCommonProps, EcsRunTaskStateBase } from './run-task-base';
+
+/**
+ * Properties to run an ECS task on EC2 in StepFunctions and ECS
+ */
+export interface EcsEc2RunTaskProps extends EcsRunTaskCommonProps {
+  /**
+   * In what subnets to place the task's ENIs
+   *
+   * (Only applicable in case the TaskDefinition is configured for AwsVpc networking)
+   *
+   * @default - Private subnets
+   */
+  readonly subnets?: ec2.SubnetSelection;
+
+  /**
+   * Existing security group to use for the task's ENIs
+   *
+   * (Only applicable in case the TaskDefinition is configured for AwsVpc networking)
+   *
+   * @default - A new security group is created
+   */
+  readonly securityGroup?: ec2.ISecurityGroup;
+
+  /**
+   * Placement constraints
+   *
+   * @default  - None
+   */
+  readonly placementConstraints?: ecs.PlacementConstraint[];
+
+  /**
+   * Placement strategies
+   *
+   * @default - None
+   */
+  readonly placementStrategies?: ecs.PlacementStrategy[];
+}
+
+/**
+ * Run an ECS/EC2 Task in a Step Functions workflow
+ */
+export class EcsEc2RunTask extends EcsRunTaskStateBase {
```

Comment:
Would the bind pattern work better here instead?

```ts
new EcsRunTask(..., {
  ...
  target: new Ec2LaunchTarget({
    cluster: ...,
    ...
  });
});
```

Reply:
I tinkered with this a little bit and while the bind pattern would work, I'm of the opinion that the current modeling is clearer. We do use the bind pattern internally to assemble the configuration.

The parameters although different work well with the bind. The wiring of the network configuration for each launch type and the current modeling in the `aws-ecs` module however, is what made me pause.

Reply:
> I'm of the opinion that the current modeling is clearer

How come? What's different/special here as opposed to where we've used them?

> The wiring of the network configuration for each launch type and the current modeling in the aws-ecs module however, is what made me pause.

Can you expand? It's not clear what you mean here.

Reply:
This will also mean that the `EcsTaskBase` class is no longer needed, which is a nice side effect.

My general rule of thumb is - if there are a few properties that change based on a choice, the bind pattern works well. However, if almost everything changes based on a choice, the subclass pattern works well. The `TaskStateBase` is a good example of the latter, this feels to fit more on the former.

This is, of course, a rule of thumb and not a rule.

Reply:
> > I'm of the opinion that the current modeling is clearer
> 
> How come? What's different/special here as opposed to where we've used them?
> 
> > The wiring of the network configuration for each launch type and the current modeling in the aws-ecs module however, is what made me pause.
> 
> Can you expand? It's not clear what you mean here.

I'll walk that back about the network configuration, we can do that in the bind.

As for what made me pause:
* The way it's currently modeled is familiar to the way it's modeled in the `aws-ecs` module.
* Feels strange to add an `IEcsLaunchTarget` or similar in this module. Feels like it should belong in the `aws-ecs` module.
* The current implementations do split these off, but I don't think it's a big win to merge them. If we expect lots more launch types to pop up, I'd view this differently.

Having said all that, I'm just going to go ahead and implement the bind and see what that looks like. If we like it more let's go with it.

Reply:
I think bind is the way to go here as well. I think its appropriate that `IEcsLaunchTarget` lives here and not in `aws-ecs` since its not really a concept/resource in ecs that would be used outside of sfn as far as I can tell. I guess it could potentially be modeled that way for patterns etc but currently it isn't.

Reply:
done